nerfstudio/data/dataparsers/nerfstudio_dataparser.py

```diff
@@ -458,6 +482,70 @@ def _load_3D_points(self, ply_file_path: Path, transform_matrix: torch.Tensor, s
         }
         return out
 
+    @staticmethod
+    def _align_points_to_target_plane(points: torch.Tensor, target_normal: torch.Tensor):
+        """Aligns a set of 3D points (in homogeneous coordinates) to a target plane defined by its normal vector.
+
+        Args:
+            points: A torch tensor of shape (n, 4) representing the 3D points in homogeneous coordinates.
+            target_normal: A torch tensor of shape (3, ) representing the normal vector of the target plane.
+
+        Returns:
+            A tuple containing:
+            - aligned_points: The 3D points aligned to the target plane as a torch tensor of shape (n, 4).
+            - alignment_matrix: The 4x4 alignment matrix used for alignment.
+        """
+        points_xyz = points[:, :3]  # Shape: (n, 3)
+
+        # Calculate the centroid (mean of points)
+        centroid = torch.mean(points_xyz, dim=0)  # Shape: (3,)
+
+        # Center the points around the centroid
+        centered_points = points_xyz - centroid  # Shape: (n, 3)
+
+        # Perform SVD
+        _, _, vh = torch.linalg.svd(centered_points)  # vh shape: (3, 3)
+
+        # The last right singular vector is the normal to the plane
+        normal = vh[-1]  # Shape: (3,)
+
+        # Calculate the rotation axis and angle
+        rotation_axis = torch.cross(normal, target_normal)  # Shape: (3,)
+        rotation_axis_norm = torch.norm(rotation_axis)
+
+        if rotation_axis_norm != 0:
```

Comment:
I would assume there would be some numerical instability when rotation_axis_norm is very close to zero? Does it make sense to use a more tolerable condition such as `rotation_axis_norm < 1e-6` or so?